File: data/code_pairs/cpp/1382.cpp
#include <iostream>

void loop3_F77(int N, double* x, double* y, double a) {
    for (int i = 0; i < N; ++i) {
        y[i] = y[i] + a * x[i];
    }
}

void loop3_F77Overhead(int N, double* x, double* y, double a) {
    // This function intentionally left empty, similar to the Fortran version.
}

File: data/code_pairs/fortran/1382.f90
SUBROUTINE loop3_F77(N, x, y, a)
      INTEGER N
      REAL*8 x(N), y(N), a
      INTEGER i

      DO i=1,N
          y(i) = y(i) + a*x(i)
      END DO
      RETURN
      END

      SUBROUTINE loop3_F77Overhead(N, x, y, a)
      INTEGER N
      REAL*8 x(N), y(N), a
      RETURN
      END